Tests/LootLockerTestUtils/LootLockerTestConfigurationUser.cs

```diff
@@ -0,0 +1,177 @@
+using LootLocker;
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+using static LootLockerTestConfigurationUtils.Auth;
+
+namespace LootLockerTestConfigurationUtils
+{
+    public class LootLockerTestUser
+    {
+        #region Statics
+        private static LootLockerTestUser _activeUser = null;
+        public static bool isSignedIn => _activeUser != null;
+        private static bool isSignInInProgress = false;
+
+        private static List<Action<bool /*success*/, string /*errorMessage*/, LootLockerTestUser /*User*/>> listeners = new List<Action<bool, string, LootLockerTestUser>>();
+
+        private static void InvokeListenersAndClear(bool success, string errorMessage)
+        {
+            if (string.IsNullOrEmpty(errorMessage))
+            {
+                Debug.LogWarning(errorMessage);
+            }
+
+            foreach (var listener in listeners)
+            {
+                listener?.Invoke(success, errorMessage, _activeUser);
+            }
+            listeners.Clear();
+        }
+
+        public static void GetUserOrSignIn(Action<bool /*success*/, string /*errorMessage*/, LootLockerTestUser /*User*/> signInComplete)
+        {
+            // We already have an active user locally, reuse it
+            if (_activeUser != null)
+            {
+                signInComplete?.Invoke(true, "", _activeUser);
+                return;
+            }
+
+            listeners.Add(signInComplete);
+
+            // A sign in process is already in progress, simply listen for that result (this is to maybe lazy deal with concurrency)
+            if (isSignInInProgress && signInComplete != null)
+            {
+                return;
+            }
+
+            // We are the primary sign in process
+            isSignInInProgress = true;
+
+            // Generate the user info
+            var userDate = DateTime.Now.ToString("yyyy-MM-dd-HH") + "h";
+            var userName = "testrun+" + userDate;
+            var password = userName;
+            var userEmail = "sdk+ci-" + userName + "@lootlocker.com";
+
+            bool isTargetingProduction = LootLockerConfig.IsTargetingProductionEnvironment();
+            if (isTargetingProduction)
+            {
+                // Don't create new users in Production
+                //TODO: Set these if you want to run towards production locally
```

Comment:
Shouldn't be a todo. Don't know what it *should* be... But shouldn't be a todo...

Reply:
Fixed